### PR TITLE
NetworkInterface: add enable ip forwarding function

### PR DESCRIPTION
**Pull request by mcgov**
For DPDK l3fwd test we need a function to enable ip forwarding. We have a similar function to enable accelerated networking, unsure how to best implement this in a production environment.

### CODE REVIEW

**Comment by mcgov** [comment]
After talking to Chi, I'm going to modify this to resolve the NIC in the VM to the Azure resource name before enabling forwarding

File: lisa/sut_orchestrator/azure/features.py [hunk 1]
```diff
@@ -651,6 +651,48 @@ def _initialize(self, *args: Any, **kwargs: Any) -> None:
...
+        vm = get_vm(azure_platform, self._node)
+        for nic in vm.network_profile.network_interfaces:
+            # get nic name from nic id
+            # /subscriptions/[subid]/resourceGroups/[rgname]/providers
+            # /Microsoft.Network/networkInterfaces/[nicname]
+            nic_name = nic.id.split("/")[-1]
+            # Since the VM nic and the Azure NIC names won't always match,
+            # allow selection by private_ip_address to resolve a NIC in the VM
+            # to an azure network interface resource.
+            if private_ip_addr and nic.private_ip_address != private_ip_addr:
```
Comment (LiliDeng) [inline lisa/sut_orchestrator/azure/features.py]: have you checked that `nic.private_ip_address` works?

Comment (mcgov) [reply lisa/sut_orchestrator/azure/features.py]: fixed! It did not work as expected, but does now.

File: lisa/sut_orchestrator/azure/features.py [hunk 2]
```diff
@@ -651,6 +651,55 @@ def _initialize(self, *args: Any, **kwargs: Any) -> None:
...
+                updated_nic.enable_ip_forwarding = enable
+                network_client.network_interfaces.begin_create_or_update(
+                    self._resource_group_name, updated_nic.name, updated_nic
+                )
+                updated_nic = network_client.network_interfaces.get(
+                    self._resource_group_name, nic_name
+                )
+                assert_that(updated_nic.enable_ip_forwarding).described_as(
+                    f"fail to set network interface {nic_name}'s accelerated "
+                    f"networking into status [{enable}]"
```
Comment (LiliDeng) [inline lisa/sut_orchestrator/azure/features.py]: update description into ip forwarding

**Review by LiliDeng — APPROVED** [review]

**Pull request merged**